src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java

```diff
@@ -1600,6 +1608,19 @@ private boolean isRepositoryVisibleToTeam(GHOrganization org, GHRepository repo,
     }
   }
 
+  private static boolean determinePrivateMode(String apiUri) {
+    try {
+      GitHub.connectToEnterpriseAnonymously(new URL(apiUri).toString()).checkApiUrlValidity();
```

Comment:
this look like an ugly and non intuitive way of saying - do not do a check if the URL is `null` however it also performs the check on github.com which would seem to be incorrect as this is only available in GHE?

```suggestion
    if (apiURL == null || apiURL.equals(GitHubServerConfig.GITHUB_URL)) {
      return false;
    }
    try {
      GitHub.connectToEnterpriseAnonymously(apiUri).checkApiUrlValidity();
```